packages/plugins/parser-cache/src/index.ts

```diff
@@ -2,9 +2,19 @@ import { Plugin } from '@envelop/types';
 import { DocumentNode, Source } from 'graphql';
 import lru from 'tiny-lru';
 
+interface Cache<T> {
+  get(key: string): T | undefined;
+  set(key: string, value: T): void;
+}
+
+export type DocumentCache = Cache<DocumentNode>;
+export type ErrorCache = Cache<Error>;
+
 export type ParserCacheOptions = {
   max?: number;
   ttl?: number;
+  documentCache?: DocumentCache;
+  errorCache?: ErrorCache;
```

Comment:
I don't like that one can provide `max` and `ttl` together with `documentCache` and `errorCache`. `max` and `ttl` should actually be config options that should be provided to the `documentCache` or `errorCache` "constructor"-call 🤔 .

In order to avoid a breaking change I would say that we deprecate the `max` and `ttl` options with a deprecated comment: https://www.typescriptlang.org/play#example/jsdoc-deprecated and also print a `console.warn`.

@dotansimha @dios-david What do you think?

Reply:
Yeah I found this weird as well, but I did not want to introduce breaking changes.
What do you think the best would be in long term? Removing `max` and `ttl`, and if someone wants to create a cache with a custtom max/ttl value then they can just create a tiny-lru cache and pass it in as e.g. `documentCache`?

Reply:
> What do you think the best would be in long term? Removing max and ttl, and if someone wants to create a cache with a custom max/ttl value then they can just create a tiny-lru cache and pass it in as e.g. documentCache?

Exactly this!

Reply:
Now that I'm thinking, there is no other changes here anyway, so why don't we just remove max/ttl in this PR and bump the major version of this plugin? 🤔

Reply:
@dios-david Fine with me, lets wait what @dotansimha says :)